frontend/src/metabase/admin/people/hooks/use-people-query.js

```diff
@@ -0,0 +1,73 @@
+import { useState, useMemo, useCallback, useEffect } from "react";
+import { SEARCH_DEBOUNCE_DURATION } from "metabase/lib/constants";
+import { USER_STATUS } from "../constants";
+
+const MIN_SEARCH_LENGTH = 2;
+
+export const usePagination = (initialPage = 0) => {
+  const [page, setPage] = useState(initialPage);
+
+  const handleNextPage = useCallback(() => setPage(prev => prev + 1), [
+    setPage,
+  ]);
+  const handlePreviousPage = useCallback(() => setPage(prev => prev - 1), [
+    setPage,
+  ]);
+
+  return {
+    handleNextPage,
+    handlePreviousPage,
+    setPage,
+    page,
+  };
+};
+
+// NOTE: EntityLoader is wrapped with PaginationState hoc, however,
+// it is not the best place to store pagination state since we might want to
+// change it from the ancestors, for instance, when we change list filter props.
+// If users change any filters, we should reset the page state.
+export const usePeopleQuery = pageSize => {
+  const { handleNextPage, handlePreviousPage, setPage, page } = usePagination();
+
+  const [status, setStatus] = useState(USER_STATUS.active);
+  const [searchInputValue, setSearchInputValue] = useState("");
+
+  const [searchText, setSearchText] = useState("");
+
+  useEffect(() => {
```

Comment:
Yes, adding something like https://github.com/streamich/react-use might be a good option. I did not have a strong need here but after we start using more functional components it will be handy